packages/web/src/features/translation/TranslationProgressBar.tsx

```diff
@@ -27,17 +42,27 @@ export default function TranslationProgressBar(
     }) +
     12; // 12px for the right/end margin;
 
+  const onResize = useCallback(
+    (progressElement: HTMLDivElement) => {
+      setTextFitsInside(textElementWidth <= progressElement.offsetWidth);
+    },
+    [textElementWidth]
+  );
+  const resizeObserverRef = useRef(
+    new ResizeObserver(([progressElementObserver]) =>
```

Comment:
I'm sorry, I gave you bad advice here. I forgot that the resize observer callback has to change when the textElementWidth changes. The proper way to design that is what you had.